tests/checkup_test.go

```diff
@@ -0,0 +1,237 @@
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	testServiceAccountName                     = "rt-checkup-sa"
+	testKiagnoseConfigMapAccessRoleName        = "kiagnose-configmap-access"
+	testKiagnoseConfigMapAccessRoleBindingName = testKiagnoseConfigMapAccessRoleName
+	testConfigMapName                          = "rt-checkup-config"
+	testCheckupJobName                         = "rt-checkup"
+)
+
+var _ = Describe("Execute the checkup Job", func() {
+	var checkupServiceAccount *corev1.ServiceAccount
+	var kiagnoseConfigMapAccessRole *rbacv1.Role
+	var kiagnoseConfigMapAccessRoleBinding *rbacv1.RoleBinding
+	var configMap *corev1.ConfigMap
+	var checkupJob *batchv1.Job
+
+	BeforeEach(func() {
+		var err error
+
+		checkupServiceAccount = newServiceAccount()
+		_, err = client.CoreV1().ServiceAccounts(testNamespace).Create(context.Background(), checkupServiceAccount, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		kiagnoseConfigMapAccessRole = newKiagnoseConfigMapAccessRole()
+		_, err = client.RbacV1().Roles(testNamespace).Create(context.Background(), kiagnoseConfigMapAccessRole, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		kiagnoseConfigMapAccessRoleBinding = newKiagnoseConfigMapAccessRoleBinding()
+		_, err = client.RbacV1().RoleBindings(testNamespace).Create(
+			context.Background(),
+			kiagnoseConfigMapAccessRoleBinding,
+			metav1.CreateOptions{},
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		configMap = newConfigMap()
+		_, err = client.CoreV1().ConfigMaps(testNamespace).Create(context.Background(), configMap, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		checkupJob = newCheckupJob()
+		_, err = client.BatchV1().Jobs(testNamespace).Create(context.Background(), checkupJob, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		DeferCleanup(func() {
+			backgroundPropagationPolicy := metav1.DeletePropagationBackground
+			err = client.BatchV1().Jobs(testNamespace).Delete(
+				context.Background(),
+				testCheckupJobName,
+				metav1.DeleteOptions{PropagationPolicy: &backgroundPropagationPolicy},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.CoreV1().ConfigMaps(testNamespace).Delete(context.Background(), testConfigMapName, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.RbacV1().RoleBindings(testNamespace).Delete(
+				context.Background(),
+				testKiagnoseConfigMapAccessRoleBindingName,
+				metav1.DeleteOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.RbacV1().Roles(testNamespace).Delete(context.Background(), testKiagnoseConfigMapAccessRoleName, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = client.CoreV1().ServiceAccounts(testNamespace).Delete(context.Background(), testServiceAccountName, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	It("should complete successfully", func() {
```

Comment:
It is sad that I need to go 60 lines above to understand what should complete successfully.

Reply:
Moved the permission setup to a dedicated function.